examples/computer_vision/iris_tf_keras/train.py

```diff
@@ -0,0 +1,138 @@
+    train = pd.read_csv(TRAIN_DATA, names=DS_COLUMNS, header=0)
+    train_features, train_labels = train, train.pop(LABEL_HEADER)
+
+    # Since we're building a classifier, convert the labels in the raw
+    # dataset (0, 1, or 2) to one-hot vector encodings that we'll to
+    # construct the Sequence data loaders that Determined expects.
+    train_labels_categorical = utils.to_categorical(train_labels, num_classes=3)
+
+    return train_features.values, train_labels_categorical
+
+
+def get_test_data():
+    test = pd.read_csv(TEST_DATA, names=DS_COLUMNS, header=0)
+    test_features, test_labels = test, test.pop(LABEL_HEADER)
+    test_labels_categorical = utils.to_categorical(test_labels, num_classes=3)
+    return test_features.values, test_labels_categorical
+
+
+def main(core_context, strategy, checkpoint, continue_id, hparams):
+    # Download train and test data.
+    train_x, train_y = get_train_data()
+    validation_data = get_test_data()
+
+    # Create and compile the model within a strategy's scope.
+    with strategy.scope():
+        inputs = layers.Input(shape=(4,))
+        dense1 = layers.Dense(hparams["layer1_dense_size"])(inputs)
+        dense2 = layers.Dense(NUM_CLASSES, activation="softmax")(dense1)
+        model = models.Model(inputs=inputs, outputs=dense2)
+
+        optimizer = legacy.RMSprop(
+            lr=hparams["learning_rate"],
+            decay=hparams["learning_rate_decay"],
+        )
+
+        model.compile(
+            optimizer,
+            losses.categorical_crossentropy,
+            [metrics.categorical_accuracy],
+        )
+
+    # Create the main DeterminedCallback that connects training to the Determined cluster.
+    det_cb = det.keras.DeterminedCallback(
+        core_context,
+        checkpoint=checkpoint,
+        continue_id=continue_id,
+        # Iris epochs are very short, so we don't even bother to save checkpoints until we finish.
+        checkpoint_epochs=0,
+    )
+
+    # Also include a Determined-aware version of the Keras' TensorBoard callback.
+    tb_cb = det.keras.TensorBoard(
+        core_context, update_freq="batch", profile_batch=0, histogram_freq=1
+    )
+
+    # Call model.fit() with our callbacks.
+    model.fit(
+        x=train_x,
+        y=train_y,
+        batch_size=hparams["global_batch_size"],
+        validation_data=validation_data,
+        epochs=100,
+        callbacks=[det_cb, tb_cb],
+    )
+
+if __name__ == "__main__":
+    logging.basicConfig(level=logging.INFO, format=det.LOG_FORMAT)
+
+    info = det.get_cluster_info()
+    if info and info.task_type == "TRIAL":
+        # We are a training a trial on-cluster.
+        continue_id = info.trial.trial_id
+        checkpoint = info.latest_checkpoint
+        # Use the hparams selected by the searcher for this trial.
+        hparams = info.trial.hparams
+    else:
+        # We are either in a notebook on-cluster or off-cluster entirely.
+        continue_id = "local-train-task"
```

Comment:
what's the benefit of forcing the user to give this a name if we're not on cluster?

Reply:
I mean... the exact same mechanics apply to how checkpoints are treated.  Checkpointing still works off cluster, it just works a little differently.